### PR TITLE
switch read and write to cache

### PR DESCRIPTION
**Pull request by grosser**
@morten @jcheatham 

this will be part of the 2.0 release -> breaking backwards compatability

### Risks
 - None

### CODE REVIEW

**Comment by grosser** [comment]
@ggrossman

File: README.md [hunk 1]
```diff
@@ -11,13 +11,7 @@ Prop supports two limiting strategies:
...
 
 ```ruby
-Prop.read do |key|
-  Rails.cache.read(key)
-end
-
-Prop.write do |key, value|
-  Rails.cache.write(key, value)
-end
+Prop.cache = Rails.cache # needs read/write/increment methods
```
Comment (jcheatham) [inline README.md]: Seems like the text above this block would need to be updated as well?

Comment (jcheatham) [inline README.md]: oh, that's in the other PR... why not base on that PR?  why not update that text here?  :confused:

**Comment by jcheatham** [comment]
I'm not really clear on the desire to change the read/write approach, but I didn't see anything obviously wrong here (though there were quite a few changes, could have missed something), so :+1: for that at least

File: README.md (hunk 1 above)
Comment (grosser) [inline README.md]: I'm making a giant mud ball and then realize it's getting too big ... splitting out things :D

**Comment by grosser** [comment]
trying to have less hops for the 99% case and makes adding increment trivial :)

**Pull request merged**